### DISCUSSION TITLE
Strange installation

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#1243] [Q&A] [answered]
I have created and installed a module to my site and noticed that two records where created in the Modules table.

![image](https://user-images.githubusercontent.com/9447612/115050221-d3995580-9edb-11eb-8d6d-20c564d6df0e.png)

the database shows

![image](https://user-images.githubusercontent.com/9447612/115050385-fdeb1300-9edb-11eb-8903-4558d0433983.png)

on the server only 6 files installed:

![image](https://user-images.githubusercontent.com/9447612/115050832-7651d400-9edc-11eb-92f2-0ce9bc529884.png)

is this correct?

**Comment by hishamco** [comment]
This is weird, seems the namespace is matter, hope to find time to validate this

**Reply by leigh-pointer** [reply]
@hishamco indeed very strange.  Infact this is happening with another module that I have created.
Deleting the moduleDef "Controls" appears not to work but the record is removed and then added again.

**Reply by hishamco** [reply]
It may need a little investigation unless @sbwalker replied quickly ;)

**Reply by leigh-pointer** [reply]
Investigating further is appears that the extra ModDef that is being registered is the namespace to the folder with my razor components.

**Reply by leigh-pointer** [reply]
Ok fixed the issue if it is one? 
All of my Razor components now inherit from ModuleControlBase this ensures Oqtane that they are part of the Module.

ModuleDefinitionRepository.cs (line 207)

```
// determine if this module implements IModule
Type moduletype = assembly
    .GetTypes()
    .Where(item => item.Namespace != null)
    .Where(item => item.Namespace == modulecontroltype.Namespace || item.Namespace.StartsWith(modulecontroltype.Namespace + "."))
    .FirstOrDefault(item => item.GetInterfaces().Contains(typeof(IModule)));
if (moduletype != null)
{
    // get property values from IModule
    var moduleobject = Activator.CreateInstance(moduletype) as IModule;
    moduledefinition = moduleobject.ModuleDefinition;
}
else
{
    // set default property values
    moduledefinition = new ModuleDefinition
    {
        Name = Utilities.GetTypeNameLastSegment(modulecontroltype.Namespace, 0),
        Description = "Manage " + Utilities.GetTypeNameLastSegment(modulecontroltype.Namespace, 0),
        Categories = ((qualifiedModuleType.StartsWith("Oqtane.Modules.Admin.")) ? "Admin" : "")
    };
}

```
because some of the conditions where not met part of the module is installed as a managed module.

**Comment by sbwalker** [comment]
This is by design. Modules must inherit from ModuleBase and be organized by namespace. You can either specify @namespace explicitly in your razor components... or Blazor will create a namespace for your components based on the folder structure and filename. Many different modules can exist in a single DLL. If you have razor components that you want to inherit from ModuleBase but not be treated as modules ( ie. shared components ) then you should use @attribute [OqtaneIgnore] in your component. This is how all of the components in the /Controls folder are configured,.

**Reply by leigh-pointer** [reply]
Ok got it.  Would it not, as a suggestion, update the description with a more rich description., especially from where it is found?

**Comment by leigh-pointer** [comment, marked as answer]
Applied the `@attribute [OqtaneIgnore]` to the razor files.
